engine/src/main/scala/cromwell/engine/io/nio/NioFlow.scala

```diff
@@ -105,30 +106,68 @@ class NioFlow(parallelism: Int,
     ()
   }
 
-  private def readAsString(read: IoContentAsStringCommand) = IO {
-    new String(
-      read.file.limitFileContent(read.options.maxBytes, read.options.failOnOverflow),
-      StandardCharsets.UTF_8
-    )
-      .replaceAll("\\r\\n", "\\\n")
+  private def readAsString(command: IoContentAsStringCommand): IO[String] = {
+    def checkHash(value: String, fileHash: FileHash): IO[ChecksumResult] = {
+      // Disable checksum validation if failOnOverflow is false. We might not read
+      // the entire stream, in which case the checksum will definitely fail.
+      // Ideally, we would only disable checksum validation if there was an actual
+      // overflow, but we don't know that here.
+      if (!command.options.failOnOverflow) return IO.pure(ChecksumSuccess())
```

Comment:
TOL wondering if maybe there shouldn't be a `ChecksumSkipped` if we're ever debugging a problem and take too much reassurance that data was correctly downloaded from`ChecksumSuccess`

Reply:
I like this. A added it, though I didn't add any behavior to distinguish it from `ChecksumSuccess`. Still expresses the intent better, and creates the opportunity to diverge the behavior in the future.